test/xbgpu/test_engine.py

```diff
@@ -614,7 +614,9 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
                 np.testing.assert_equal(expected_output, device_result)
             incomplete_accums_counters.append(incomplete_accums_counter)
 
-        xpipelines: list[XPipeline] = [pipeline for pipeline in xbengine._pipelines if isinstance(pipeline, XPipeline)]
+        xpipelines: list[XPipeline] = [
+            pipeline for pipeline in xbengine._pipelines if isinstance(pipeline, XPipeline)  # type: ignore
```

Comment:
Hmm, in theory mypy should have enough information that it doesn't need to ignore. What is the error?